contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,199 @@
+import "../libraries/SafeERC20.sol";
+
+import "../interfaces/IERC20.sol";
+import "./interfaces/IJoeERC20.sol";
+import "./interfaces/IJoePair.sol";
+import "./interfaces/IJoeFactory.sol";
+
+import "../boringcrypto/BoringOwnable.sol";
+
+interface IMasterChef {
+    struct PoolInfo {
+        IJoeERC20 lpToken; // Address of LP token contract.
+        uint256 allocPoint; // How many allocation points assigned to this pool. JOE to distribute per block.
+        uint256 lastRewardTimestamp; // Last block number that JOE distribution occurs.
+        uint256 accJoePerShare; // Accumulated JOE per share, times 1e12. See below.
+    }
+
+    function totalAllocPoint() external view returns (uint256);
+
+    function joePerSec() external view returns (uint256);
+}
+
+contract FarmLens is BoringOwnable {
+    using SafeMath for uint256;
+
+    address public joe; // 0x6e84a6216eA6dACC71eE8E6b0a5B7322EEbC0fDd;
+    address public wavax; // 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
+    IJoeFactory public joeFactory; // IJoeFactory(0x9Ad6C38BE94206cA50bb0d90783181662f0Cfa10);
+    IMasterChef public chef; //0xd6a4F121CA35509aF06A0Be99093d08462f53052
+    IMasterChef public chefv3; //0x188bED1968b795d5c9022F6a0bb5931Ac4c18F00
+
+    constructor(
+        address joe_,
+        address wavax_,
+        IJoeFactory joeFactory_,
+        IMasterChef chef_,
+        IMasterChef chefv3_
+    ) public {
+        joe = joe_;
+        wavax = wavax_;
+        joeFactory = IJoeFactory(joeFactory_);
+        chef = chef_;
+        chefv3 = chefv3_;
+    }
+
+    function getAvaxPrice() public view returns (uint256) {
+        uint256 priceFromWavaxUsdt = _getAvaxPrice(IJoePair(address(0xeD8CBD9F0cE3C6986b22002F03c6475CEb7a6256))); // 18
+        uint256 priceFromWavaxUsdc = _getAvaxPrice(IJoePair(address(0x87Dee1cC9FFd464B79e058ba20387c1984aed86a))); // 18
+        uint256 priceFromWavaxDai = _getAvaxPrice(IJoePair(address(0xA389f9430876455C36478DeEa9769B7Ca4E3DDB1))); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdt.add(priceFromWavaxUsdc).add(priceFromWavaxDai); // 18
+        uint256 avaxPrice = sumPrice / uint256(3); // 18
+        return avaxPrice; // 18
+    }
+
+    function _getAvaxPrice(IJoePair pair) public view returns (uint256) {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        if (pair.token0() == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(pair.token1())); // 18
+            return (reserve1.mul(uint256(1e18))) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(pair.token0())); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function getPriceInUSD(address tokenAddress) public view returns (uint256) {
+        return getAvaxPrice() / getPriceInAvax(tokenAddress); // 18
+    }
+
+    // Need to be aware of decimals here, not always 18, it depends on the token
```

Comment:
Use natspec and `@dev` here: https://docs.soliditylang.org/en/v0.5.10/natspec-format.html